src/test/scala/co/teapot/tempest/server/TempestSQLDatabaseClientSpec.scala

```diff
@@ -1,14 +1,77 @@
 package co.teapot.tempest.server
 
 import co.teapot.tempest.typedgraph.Node
-import co.teapot.tempest.util.ConfigLoader
-import co.teapot.tempest.{Node => ThriftNode, SQLException}
-import com.zaxxer.hikari.pool.HikariPool.PoolInitializationException
-import org.scalatest.{FlatSpec, Matchers}
+import co.teapot.tempest.{SQLException, InvalidArgumentException, Node => ThriftNode}
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers, Suite}
+
+trait H2DatabaseBasedTest {
+  this: Suite =>
+  val testConfig: DatabaseConfig = new H2DatabaseConfig
+
+  protected def createTempestSQLDatabaseClient(): TempestSQLDatabaseClient = {
+    new TempestSQLDatabaseClient(testConfig)
+  }
+}
+
+trait SyntheticDatabaseData extends BeforeAndAfterEach {
+  this: H2DatabaseBasedTest with Suite =>
+
+  def prepareTables(connection: java.sql.Connection): Unit = {
+    // TODO: Perhaps we should load this directly from the config/csv files
```

Comment:
Having everything in code has the benefit that it's easier to quickly lookup what's the shape of the data.